chapters/sampling-randomization-power.tex

```diff
@@ -1,427 +1,759 @@
-At the end of the section, 
-we provide a do-file that provides a concrete example of how to implement these principles. 
-
-Stata, like most statistical software, uses a \textbf{pseudo-random number generator}.
-Basically, it has a pre-calculated really long ordered list of numbers with the property that
-knowing the previous one gives you precisely zero information about the next one, i.e. a list of random numbers.
-Stata uses one number from this list every time it has a task that is non-deterministic.
-In ordinary use, it will cycle through these numbers starting from a fixed point
-every time you restart Stata, and by the time you get to any given script,
-the current state and the subsequent states will be as good as random.\sidenote{
-  \url{https://www.stata.com/manuals14/rsetseed.pdf}}
-However, for true reproducible randomization, we need two additional properties:
-we need to be able to fix the starting point so we can come back to it later;
-and we need to ensure that the starting point is independently random from our process.
-In Stata, this is accomplished through three command concepts:
+This section introduces the strict rules:
+these are non-negotiable (but thankfully simple).
+Stata, like most statistical software, uses a \textbf{pseudo-random number generator}
+which, in ordinary research use, is as good as random.\sidenote{
+	\url{https://dimewiki.worldbank.org/Randomization\_in\_Stata}}
+However, for \textit{reproducible} randomization, we need two additional properties:
+we need to be able to fix the sequence of numbers generated and
+ensure that the first number is independently randomized.
+In Stata, this is accomplished through three concepts:
 \textbf{versioning}, \textbf{sorting}, and \textbf{seeding}.
+We again use Stata in our examples,
+but the same principles translate to all other programming languages.
 
+% rule 1: versioning
 \textbf{Versioning} means using the same version of the software each time you run the random process.
 If anything is different, the underlying list of random numbers may have changed,
 and it will be impossible to recover the original result.
 In Stata, the \texttt{version} command ensures that the list of random numbers is fixed.\sidenote{
-At the time of writing we recommend using \texttt{version 13.1} for backward compatibility;
-the algorithm used to create this list of random numbers was changed after Stata 14 but its improvements do not matter in practice.
-You will \textit{never} be able to reproduce a randomization in a different software,
-such as moving from Stata to R or vice versa.}
+	At the time of writing, we recommend using \texttt{version 13.1} for backward compatibility;
+	the algorithm used to create this list of random numbers was changed after Stata 14 but the improvements do not matter in practice.
+	You will \textit{never} be able to reproduce a randomization in a different software,
+	such as moving from Stata to R or vice versa.}
```

Comment:
@bbdaniels suggestion: long side note, should we put int into the text?

I think we can maybe solve it by removing the sentence:
> You will \textit{never} be able to reproduce a randomization in a different software,
such as moving from Stata to R or vice versa.}

Reply:
removed in 6760d00b22580019fd278d23a676f8bfa38beb9d